rmw_fastrtps_cpp/src/rmw_client.cpp

```diff
@@ -222,6 +222,9 @@ rmw_create_client(const rmw_node_t * node,
     delete info;
   }
 
+  if (rmw_client->service_name != nullptr) {
+    rmw_free(const_cast<char *>(rmw_client->service_name));
```

Comment:
Please set `rmw_client->service_name` to `nullptr` afterwards. While the variable isn't being used anymore it makes debugging much easier when the pointer isn't left with the same value but pointing to deallocated memory.

Same below.